### PR TITLE
Feature/deprecate error handling

### PR DESCRIPTION
**Pull request by will-zegers-bc**

### CODE REVIEW

File: aruco_ros/src/aruco_detector.cpp [hunk 1]
```diff
@@ -232,62 +240,65 @@ class ArucoSimple
...
     static tf::TransformBroadcaster br;
+    bool hasTransformToReference;
 
     tf::Matrix3x3(transform.getRotation()).getRPY(roll, pitch, yaw);
 
+    // Get TF between camera and reference frame
+    tf::StampedTransform cameraToReference;
+    cameraToReference.setIdentity();
+
+    if ( reference_frame == camera_frame ) {
```
Comment (jrad77) [inline aruco_ros/src/aruco_detector.cpp]: super nitpick - no space between the parens

File: aruco_ros/src/aruco_detector.cpp [hunk 2]
```diff
@@ -232,62 +240,65 @@ class ArucoSimple
...
-      error_message = arucoMsg.CODE_TWISTED_MESSAGE;
-      error_condition |= arucoMsg.CODE_TWISTED;
+      error_message = aruco_msgs::Marker::CODE_TWISTED_MESSAGE;
+      error_condition |= aruco_msgs::Marker::CODE_TWISTED;
     }
     if (abs(remainder(yaw - expected_yaw, 2*M_PI))  > M_PI/2.){
-      error_message = arucoMsg.CODE_UPSIDE_DOWN_MESSAGE;
-      error_condition |= arucoMsg.CODE_UPSIDE_DOWN;
+      error_message = aruco_msgs::Marker::CODE_UPSIDE_DOWN_MESSAGE;
+      error_condition |= aruco_msgs::Marker::CODE_UPSIDE_DOWN;
```
Comment (jrad77) [inline aruco_ros/src/aruco_detector.cpp]: the error messages are overwritten, but the conditions are stacked - is that intentional?

File: aruco_ros/src/aruco_detector.cpp [hunk 3]
```diff
@@ -316,45 +327,66 @@ class ArucoSimple
     arucoMsg.error_code = error_condition;
     arucoMsg.error_message = error_message;
     arucoMsg.pose = poseMsg;
+
+    // Include the corner points of the marker, so we an reconstruct its position in the UI
+    for (int i = 0; i < 4; ++i) {
+      aruco_msgs::Corner corner;
+      corner.x = marker[i].x;
```
Comment (jrad77) [inline aruco_ros/src/aruco_detector.cpp]: is there any way to confirm that the are four elements in marker?

File: aruco_ros/src/aruco_detector.cpp [hunk 4]
```diff
@@ -316,45 +327,66 @@ class ArucoSimple
...
+      corner.y = marker[i].y;
+      arucoMsg.corners.push_back(corner);
+    }
     pose_pub.publish(arucoMsg);
   }
 
   void image_callback(const sensor_msgs::ImageConstPtr& msg)
   {
+    unsigned int error_code;
```
Comment (jrad77) [inline aruco_ros/src/aruco_detector.cpp]: is this unused?

File: aruco_ros/src/aruco_detector.cpp [hunk 5]
```diff
@@ -316,45 +327,66 @@ class ArucoSimple
...
     if(cam_info_received)
     {
       cv_bridge::CvImagePtr cv_ptr;
-
       try
       {
         cv_ptr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::RGB8);
         inImage = cv_ptr->image;
 
+        if (is_camera_rotated) { cv::rotate(inImage, inImage, cv::ROTATE_90_CLOCKWISE); }
```
Comment (jrad77) [inline aruco_ros/src/aruco_detector.cpp]: multiple lines please

Comment (jrad77) [reply aruco_ros/src/aruco_detector.cpp]: also, in in place rotations work?

File: aruco_ros/src/aruco_detector.cpp (hunk 2 above)
Comment (will-zegers-bc) [reply aruco_ros/src/aruco_detector.cpp]: Yea, now that you bring it up it doesn't make all that much sense to stack the conditions if we're just logging one message. This was just carried over from what they were doing previously, but I think handling one error condition makes things a little more clear

File: aruco_ros/src/aruco_detector.cpp (hunk 5 above)
Comment (will-zegers-bc) [reply aruco_ros/src/aruco_detector.cpp]: My thoughts at first too. Looking at the OCV function, looks like they're copying data behind the scenes

File: aruco_ros/src/aruco_detector.cpp (hunk 3 above)
Comment (will-zegers-bc) [reply aruco_ros/src/aruco_detector.cpp]: assert()? Marker is just an extension of a cv::point2f vector, so we can just assert the the size of the vec is 4

**Pull request merged**